src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetCSVGenerator.java

```diff
@@ -60,7 +63,14 @@ public File generateMultiInstanceFile(final Map<Integer, List<ObservationUnitRow
 	String[] getColumnValues(final ObservationUnitRow row, final List<MeasurementVariable> subObservationSetColumns) {
 		final List<String> values = new LinkedList<>();
 		for (final MeasurementVariable column : subObservationSetColumns) {
-			values.add(row.getVariables().get(column.getName()).getValue());
+			if (row.getEnvironmentVariables().containsKey(column.getName()) && (
+				column.getVariableType().getId() == VariableType.ENVIRONMENT_DETAIL.getId()
+					|| column.getVariableType().getId() == VariableType.STUDY_CONDITION.getId())) {
```

Comment:
Create static list of variable types here for readability

Reply:
Also here, related to comment in abstracting in `ObservationUnitRow `class where value comes from

Reply:
> Create static list of variable types here for readability

This is done.